plugins/poll/assets/javascripts/widgets/discourse-poll.js.es6

```diff
@@ -415,6 +422,211 @@ createWidget("discourse-poll-info", {
+          if (!el) {
+            const container = document.createElement("div");
+            container.classList.add("poll-grouped-pie-container");
+
+            const label = document.createElement("label");
+            label.classList.add("poll-pie-label");
+            label.textContent = result.grouped_results[chartIdx].group;
+
+            const canvas = document.createElement("canvas");
+            canvas.classList.add(`poll-grouped-pie-${attrs.id}`);
+            canvas.id = canvasId;
+
+            container.appendChild(label);
+            container.appendChild(canvas);
+            parent.appendChild(container);
+            // eslint-disable-next-line
+            new Chart(canvas.getContext("2d"), chartConfig);
+          } else {
+            // eslint-disable-next-line
+            Chart.helpers.each(Chart.instances, function(instance) {
+              if (instance.chart.canvas.id === canvasId && el.$chartjs) {
+                instance.destroy();
+                // eslint-disable-next-line
+                new Chart(el.getContext("2d"), chartConfig);
+              }
+            });
+          }
+        }
+      });
+    });
+    return "";
+  },
+
+  click(e) {
+    let select = $(e.target).closest("select");
+    if (select.length) {
+      this.sendWidgetAction("refreshCharts", select[0].value);
+    }
+  }
+});
+
+function clearPieChart(id) {
+  let el = document.querySelector(`#poll-results-chart-${id}`);
+  el && el.parentNode.removeChild(el);
+}
+
+createWidget("discourse-poll-pie-canvas", {
+  tagName: "canvas.poll-results-canvas",
+
+  buildAttributes(attrs) {
+    return {
+      id: `poll-results-chart-${attrs.id}`
+    };
+  }
+});
+
+createWidget("discourse-poll-pie-chart", {
+  tagName: "div.poll-results-chart",
+  html(attrs) {
+    const contents = [];
+
+    if (!attrs.showResults) {
+      clearPieChart(this.attrs.id);
+      return contents;
+    }
+
+    let btn;
+    let chart;
+    if (attrs.groupResults && attrs.groupableUserFields.length > 0) {
+      btn = h("div", [
+        this.attach("button", {
+          className: "btn-default poll-group-by-toggle",
+          label: "poll.ungroup-results.label",
+          title: "poll.ungroup-results.title",
+          icon: "far-eye-slash",
+          action: "toggleGroupedPieCharts"
+        })
+      ]);
+      chart = this.attach("discourse-poll-grouped-pies", attrs);
+      clearPieChart(this.attrs.id);
+    } else {
+      if (attrs.groupableUserFields.length > 0) {
+        btn = this.attach("button", {
+          className: "btn-default poll-group-by-toggle",
+          label: "poll.group-results.label",
+          title: "poll.group-results.title",
+          icon: "far-eye",
+          action: "toggleGroupedPieCharts"
+        });
+      }
+      const data = attrs.poll.options.map(o => o.votes);
+      const labels = attrs.poll.options.map(o => o.html);
+      loadScript("/javascripts/Chart.min.js").then(() => {
+        later(() => {
```

Comment:
can you explain this later please?

Should more likely be a `schedule("afterRender", () => ` I think